Pod/Classes/BONTag.m

```diff
@@ -0,0 +1,235 @@
+{
+    NSString *description = [NSString stringWithFormat:@"<%@: %p, startTag: %@, endTag: %@, textable: <%@: %p>>", NSStringFromClass(self.class), self, self.startTag, self.endTag, NSStringFromClass(self.textable.class), self.textable];
+
+    return description;
+}
+
+#pragma mark - Tag Matching
+
++ (NSArray *)escapedRangesInString:(NSString **)string withTags:(NSArray *)tags stripEscapeCharacters:(BOOL)stripEscapeCharacters
+{
+    NSMutableArray *escapedRanges = [NSMutableArray array];
+
+    NSString *theString = *string;
+
+    NSRange searchRange = NSMakeRange(0, theString.length);
+
+    // Iterate over the string finding each escape in order until there are no more escape strings
+    while (YES) {
+        BONTag *nextTag;
+        NSRange nextEscapeRange;
+        NSRange nextEscapedTagRange;
+
+        // Find the next escape string
+        for (BONTag *tag in tags) {
+            NSRange escapeRange = [theString rangeOfString:tag.escapeString options:0 range:searchRange];
+
+            if (escapeRange.location != NSNotFound) {
+                if (!nextTag || (escapeRange.location < nextEscapeRange.location)) {
+                    // Check if this character is escaping a start or end tag
+                    for (NSString *tagString in @[ tag.startTag, tag.endTag ]) {
+                        // Check if there is room for this tag to exist after the escape character
+                        if ((NSInteger)(NSMaxRange(escapeRange) + tagString.length) < theString.length) {
+                            // Check if the following characters are actually the tag
+                            NSRange potentialEscapedTagRange = NSMakeRange(NSMaxRange(escapeRange), tagString.length);
+                            if ([[theString substringWithRange:potentialEscapedTagRange] isEqualToString:tagString]) {
+                                nextTag = tag;
+                                nextEscapeRange = escapeRange;
+                                nextEscapedTagRange = potentialEscapedTagRange;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        if (!nextTag) {
+            break;
+        }
+
+        NSRange range = NSMakeRange(nextEscapeRange.location, nextEscapedTagRange.length + nextEscapeRange.length);
+
+        if (stripEscapeCharacters) {
+            theString = [theString stringByReplacingOccurrencesOfString:nextTag.escapeString withString:@"" options:0 range:nextEscapeRange];
+            range = NSMakeRange(nextEscapeRange.location, nextEscapedTagRange.length);
+        }
+
+        [escapedRanges addObject:[NSValue valueWithRange:range]];
+
+        searchRange = NSMakeRange(NSMaxRange(range), [theString length] - NSMaxRange(range));
+    }
+
+    *string = theString;
+
+    return escapedRanges;
+}
+
++ (NSRange)findNextString:(NSString *)string inString:(NSString *)stringToSearch ignoringRanges:(NSArray *)escapedRanges range:(NSRange)range
```

Comment:
#nth these class methods are ripe for isolated testing. You can expose them in `BONTag_Testing.h` or in a category in the test header, if you want. Up to you, but it could help to clarify how they work.